### PR TITLE
Plugin process:  in Linux count open file descriptors.

### PR DESCRIPTION
**Pull request by manuelluis**
Add the option CollectFileDescriptor since it's count the files in ``/proc/pid/fd/*`` and it's add some overhead.

### CODE REVIEW

File: src/processes.c [hunk 1]
```diff
@@ -592,6 +602,18 @@ static int ps_config (oconfig_item_t *ci)
 			ps_list_register (c->values[0].value.string,
 					c->values[1].value.string);
 		}
+		else if (strcasecmp (c->key, "CollectFileDescriptor") == 0)
+		{
+
+			if ((c->values_num != 1)
```
Comment (octo) [inline src/processes.c]: Can you please use `cf_util_get_boolean()` here? It will do the validation and logging for you.

File: src/processes.c [hunk 2]
```diff
@@ -211,12 +213,16 @@ typedef struct procstat
...
 
+	unsigned long nfd;
+
 	struct procstat   *next;
 	struct procstat_entry_s *instances;
 } procstat_t;
 
 static procstat_t *list_head_g = NULL;
 
+static _Bool ps_fd = 0;
```
Comment (octo) [inline src/processes.c]: The use of this variable was not obvious to me. Can you please rename to `collect_nfd`? We use `collect_*` in many other plugins for variables of this purpose.

File: src/types.db [hunk 3]
```diff
@@ -166,6 +166,7 @@ ps_cputime		user:DERIVE:0:U, syst:DERIVE:0:U
 ps_data			value:GAUGE:0:9223372036854775807
 ps_disk_octets		read:DERIVE:0:U, write:DERIVE:0:U
 ps_disk_ops		read:DERIVE:0:U, write:DERIVE:0:U
+ps_fd		value:GAUGE:0:9223372036854775807
```
Comment (octo) [inline src/types.db]: Please reuse the `file_handles` type of the (brand-new) *fhcount* plugin.

**Comment by octo** [comment]
Thank you very much for your patch, Manuel! I've added a few nits, apart from this this is good to go, I think :)

—octo

**Comment by mfournier** [comment]
See also #836 which does basically the same thing (maybe in a more simplistic way). I've used it for a couple of months now, and it has already been pretty useful.

I don't have a strong opinion about one vs the other implementation. But definetely +1 for such a feature !

**Comment by octo** [comment]
Both look fine to me. I like the `ps_count_dir_entries()` solution in @mfournier's approach :)

**Comment by manuelluis** [comment]
Sorry , I had not seen #836. It's looks fine to me.

My concern is that is enabled by default, since it's a expensive operation if you have a lot of process and open files or connections.

**Pull request closed without merging**